modules/gateway/conn.go

```diff
@@ -24,13 +24,13 @@ func (pc peerConn) RPCAddr() modules.NetAddress {
 // communication protocol.
 func (g *Gateway) dial(addr modules.NetAddress) (net.Conn, error) {
 	dialer := &net.Dialer{
-		Timeout: dialTimeout,
-		Cancel:  g.threads.StopChan(),
+		Cancel:   g.threads.StopChan(),
+		Deadline: time.Now().Add(connStdDeadline),
+		Timeout:  dialTimeout,
 	}
 	conn, err := dialer.Dial("tcp", string(addr))
 	if err != nil {
 		return nil, err
 	}
-	conn.SetDeadline(time.Now().Add(connStdDeadline))
```

Comment:
oh, I was wrong. It's only when you are doing http transports that you shouldn't be setting a timeout on the conn itself.

I may have fixed the problem via some of the other changes, I added some bounds to the number of concurrent outbound connections the gateway will have, which I think made things a lot easier on the antfarm.

source of wrong information: gropus.google.com/forum/#!topic/golang-nuts/VncYuUsEMvY